### PR TITLE
add in more specs and docs

### PR DESCRIPTION
**Pull request by allenwyma**
# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [X] Documentation changes
- [ ] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

File: lib/ash/page/offset.ex [hunk 1]
```diff
@@ -7,7 +7,23 @@ defmodule Ash.Page.Offset do
   """
   defstruct [:results, :limit, :offset, :count, :rerun, :more?]
 
-  @type t :: %__MODULE__{}
+  @type t :: %__MODULE__{
+          results: [Ash.Resource.t()],
```
Comment (zachdaniel) [inline lib/ash/page/offset.ex:11]: `Ash.Resource.record` the `t` type refers to a module

**Review by zachdaniel — CHANGES_REQUESTED** [review]

File: lib/ash/page/offset.ex (hunk 1 above)
Comment (allenwyma) [reply lib/ash/page/offset.ex:11]: Okay.

At what point is it a Resource.t versus Resource.record?

Comment (zachdaniel) [reply lib/ash/page/offset.ex:11]: A `Resource.t` is `MyApp.Accounts.User` and `Resource.record` is `%MyApp.Accounts.User{}`

Comment (zachdaniel) [inline lib/ash/page/offset.ex:11]:
```suggestion
          results: [Ash.Resource.record()],
```

**Review by zachdaniel — APPROVED** [review]

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**